src/client/options/mod.rs

```diff
@@ -949,6 +949,13 @@ impl ClientOptions {
         Self::parse_uri(s, None).await
     }
 
+    /// Async version that is unconditionally available in tests, to simplify code for parsing
+    /// connection strings in tests.
+    #[cfg(test)]
+    pub async fn parse_async(s: impl AsRef<str>) -> Result<Self> {
```

Comment:
this is a placeholder, and should be removed in favor of the new `ClientOptions::parse_connection_string` method once that gets merged in. (see my comment below for more details)